crates/hir/src/semantics.rs

```diff
@@ -259,6 +259,10 @@ impl<'db, DB: HirDatabase> Semantics<'db, DB> {
     }
 
     pub fn to_module_def(&self, file: FileId) -> Option<Module> {
+        self.imp.to_module_def(file).next()
+    }
+
+    pub fn to_module_defs(&self, file: FileId) -> impl Iterator<Item = Module> {
```

Comment:
In the future, we should get rid of the first function. It's problematic in that it's not obvious, at the call site, that more than one result can be returned.  Although, just returening a vec in call cases seems wrong as well. Rather, the caller probably wants to pass in some kind of predicate object, which is used to select the best possible module. 

So, something like `fn guess_module_def(file: FiledId, node: &SyntaxNode)` might make sens.

Reply:
Ye was thinking about that as well a bit, getting the predicate/heuristic right seems difficult